cinder/volume/api.py

```diff
@@ -826,6 +827,43 @@ def migrate_volume_by_connector(self, ctxt, volume, connector):
         backend = host_manager.BackendState(host=dest['host'],
                                             cluster_name=dest['cluster_name'],
                                             capabilities=dest['capabilities'])
+
+        # Build required conditions for conditional update
+        expected = {'status': ('available', 'reserved'),
+                    'migration_status': self.AVAILABLE_MIGRATION_STATUS,
+                    'replication_status': (
+                        None,
+                        fields.ReplicationStatus.DISABLED,
+                        fields.ReplicationStatus.NOT_CAPABLE),
+                    'consistencygroup_id': (None, ''),
+                    'group_id': (None, '')}
+
+        expected['host'] = db.Not(volume.host)
+        filters = [~db.volume_has_snapshots_filter()]
+
+        updates = {'migration_status': 'starting',
+                   'previous_status': volume.model.status}
+
+        # When the migration of an available volume starts, both the status
+        # and the migration status of the volume will be changed.
+        # If the admin sets lock_volume flag to True, the volume
+        # status is changed to 'maintenance', telling users
+        # that this volume is in maintenance mode, and no action is allowed
+        # on this volume, e.g. attach, detach, retype, migrate, etc.
+        if lock_volume:
+            updates['status'] = db.Case(
+                [(volume.model.status == 'available', 'maintenance')],
```

Comment:
What about `reserved` here, don't we have to change to `maintenance` then, too?

Reply:
done